### PR TITLE
Try reporting diagnostics in non-strict mode

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Makes sure that the compiler will print the diagnostics even if non-strict mode is used
	- To prevent printing unexpected stuff to stdout in interactive mode, the diagnostics are printed as 'warning' level log messages in that mode. In strict mode, the messages are printed like before, without changes.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
I realised that we _do_ seem to be reporting these errors in interactive mode (even if we don't 'hit' them), at least to some extent - as seen in https://github.com/enso-org/enso/blob/develop/engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeErrorsTest.scala#L679-L695

I thought we are not.

Still, I think this could be improvement.

1. IIRC we don't yet show these messages in the IDE in any way. So it will be helpful to see them in any form before we get the proper display.
2. Even once we get it, I think it is beneficial to have these diagnostics in the logs. The users often attach logs when reporting issues - and the diagnostics may help in finding out the problems - so having them there (in the interactive mode) could make it easier to find issues.

File: engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala [hunk 1]
```diff
@@ -1126,8 +1127,9 @@ class Compiler(
           var str = fansi.Str()
           val fileLocation = diagnostic.location match {
             case Some(_) => fileLocationFromSection(diagnostic.location, source)
-            case None    => source.getPath
+            case None    => Option(source.getPath).getOrElse("<Unknown source>")
```
Comment (Akirathan) [inline engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala]: nice observation, thx

**Review by Akirathan — APPROVED** [review]

File: engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala [hunk 2]
```diff
@@ -1272,6 +1274,10 @@ class Compiler(
   def updateMetadata(sourceIr: IRModule, copyOfIr: IRModule): IRModule = {
     passManager.runMetadataUpdate(sourceIr, copyOfIr)
   }
+
+  private def printDiagnostic(message: String): Unit =
+    if (config.isStrictErrors) output.println(message)
+    else context.log(Level.WARNING, message)
```
Comment (JaroslavTulach) [inline engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala:1258]: Flooding log with [user errors](https://github.com/orgs/enso-org/discussions/8271#discussioncomment-7529419) is probably not very good idea in the long term.

**Review by JaroslavTulach — APPROVED** [review]
I don't like the `Level.WARNING` change myself, but I don't mind having it in.

**Pull request merged**